completions/make.lua

```diff
@@ -0,0 +1,145 @@
+local clink_version = require('clink_version')
+if not clink_version.supports_argmatcher_delayinit then
+    print("make.lua argmatcher requires a newer version of Clink; please upgrade.")
+    return
+end
+
+-- Table of special targets to always ignore.
+local special_targets = {
+    ['.PHONY'] = true,
+    ['.SUFFIXES'] = true,
+    ['.DEFAULT'] = true,
+    ['.PRECIOUS'] = true,
+    ['.INTERMEDIATE'] = true,
+    ['.SECONDARY'] = true,
+    ['.SECONDEXPANSION'] = true,
+    ['.DELETE_ON_ERROR'] = true,
+    ['.IGNORE'] = true,
+    ['.LOW_RESOLUTION_TIME'] = true,
+    ['.SILENT'] = true,
+    ['.EXPORT_ALL_VARIABLES'] = true,
+    ['.NOTPARALLEL'] = true,
+    ['.ONESHELL'] = true,
+    ['.POSIX'] = true,
+    ['.NOEXPORT'] = true,
+    ['.MAKE'] = true,
+}
+
+local function extract_target(line, last_line, targets, allow_pathlike)
+    -- Ignore comment lines.
+    if line:find('#', 1, true) then
+        return
+    end
+
+    -- Ignore when not a target (is this only for GNU make?).
+    if last_line:find('# Not a target') then
+        return
+    end
+
+    -- Extract possible target.
+    local p = (
+        line:match('^([^%s]+):$') or -- When target has no deps.
+        line:match('^([^%s]+): '))   -- When target has deps.
+    if not p then
+        return
+    end
+
+    -- Ignore targets with pattern
```

Comment:
Is "pattern" a typo?  Was it meant to be "percent"?

Reply:
It is pattern.
https://www.gnu.org/software/make/manual/html_node/Pattern-Intro.html